src/urllib3/util/ssl_.py

```diff
@@ -241,9 +241,10 @@ def create_urllib3_context(
         context.post_handshake_auth = True
 
     context.verify_mode = cert_reqs
-    # We do our own verification, including fingerprints and alternative
-    # hostnames. So disable it here
-    context.check_hostname = False
+    # We ask for verification here but it may be disabled in HTTPSConnection.connect
+    context.check_hostname = cert_reqs == ssl.CERT_REQUIRED
+    if hasattr(context, "hostname_checks_common_name"):
+        context.hostname_checks_common_name = False  # Python 3.7
```

Comment:
Should the `# Python ...` comment be on the `if` line to cover the whole branch?

Reply:
So the intent here is to remind us to remove this when we drop support for Python 3.6 while still making sure it's covered.

Now that I think about it more, the best thing to do is to rely on codecov telling us it's no longer covered when we drop Python 3.6.